Issue #348: Adopt the changes in 'liberty-maven-plugin' 3.0

josephca opened the pull request.

Commit:
Update to jsch 0.1.54

Commit:
Merge branch 'integration' of github.com:OpenLiberty/open-liberty-tools into integration

Commit:
Use deploy goal instead of the obsolete install-apps goal

Commit:
Revert build.gradle

cherylking commented on dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/internal/LibertyMaven.java:
```diff
@@ -187,7 +187,7 @@ public boolean isSupportedProject(IProject project, IProgressMonitor monitor) {
     @Override
     public void updateSrcConfig(IPath location, LibertyBuildPluginConfiguration config, IProgressMonitor monitor) {
-        String goal = "package liberty:install-apps";
+        String goal = "package liberty:deploy";
```
Don't you need to check which version of the liberty maven plugin is configured before deciding whether to call install-apps or deploy goal? Before 3.x, install-apps was the correct goal to call. Calling deploy instead would not work if a 2.x version of the plugin was configured.

c-koell replied:
Please consider also the other changes like create-server -> create and so on ...

josephca replied:
> Please consider also the other changes like create-server -> create and so on ...


Yes, they will be handled together.

josephca replied:
> Don't you need to check which version of the liberty maven plugin is configured before deciding whether to call install-apps or deploy goal? Before 3.x, install-apps was the correct goal to call. Calling deploy instead would not work if a 2.x version of the plugin was configured.

This update will result in some impact on Websphere Developer Tools as well, so handling both 2.x/3.x may not be an easy option. Rather, we should consider to document it to use 3.x for upcoming OLT/WDT releases (20.0.0.3 and higher)

cherylking reviewed (changes requested):
I'm not sure where the parameters for the goals are handled, but install-apps and deploy have different parameters that they expect. See the release notes for details: https://github.com/OpenLiberty/ci.maven/releases#changed-deploy-goal-and-removed-install-apps-goal

Commit:
Bump up the feature version

josephca commented:
> I'm not sure where the parameters for the goals are handled, but install-apps and deploy have different parameters that they expect. See the release notes for details: https://github.com/OpenLiberty/ci.maven/releases#changed-deploy-goal-and-removed-install-apps-goal



> I'm not sure where the parameters for the goals are handled, but install-apps and deploy have different parameters that they expect. See the release notes for details: https://github.com/OpenLiberty/ci.maven/releases#changed-deploy-goal-and-removed-install-apps-goal

 In the existing 'install-apps' goal calls, no param has been passed, thus we don't need to map the parameters.

Commit:
revert back file format change

Commit:
Update

Commit:
update

Commit:
update

Commit:
update

Commit:
update

Commit:
Merge branch 'integration' of github.com:OpenLiberty/open-liberty-tools into 348-liberty-maven-plugin_3.0
